rest/rest_request_router.h

```diff
@@ -281,9 +281,31 @@ struct RestRequestParsingContext {
         const std::type_info * type;
         std::function<void (void *) noexcept> deleter;
 
-        //ObjectEntry(const ObjectEntry &) = delete;
-        //void operator = (const ObjectEntry &) = delete;
+        ObjectEntry(const ObjectEntry &) = delete;
+        void operator = (const ObjectEntry &) = delete;
 
+        ObjectEntry(ObjectEntry && other) noexcept
+            : obj(other.obj), type(other.type), deleter(std::move(other.deleter))
+        {
+            other.obj = nullptr;
+            other.deleter = nullptr;
+        }
+
+        ObjectEntry & operator = (ObjectEntry && other) noexcept
+        {
+            ObjectEntry newMe(std::move(other));
```

Comment:
It's a common pattern to write a move and copy constructor that's noexcept without repeating code or writing a version of the move assignment that needs to clean up:
- We first move the object into a different copy, which means that the move constructor is what's called.  The move constructor is easier to write as it never needs to clean anything up, and it's noexcept.
- We then swap the new me with *this, which is again noexcept.
- Finally, we destroy newMe, and leave other to be cleaned up by the calling code.  Thus the cleanup is done automatically; no need to write code to clean up.

Since none of these can throw an exception, we have a noexcept move assignment without needing to know any details of how things are implemented.